config/reader.go

```diff
@@ -438,13 +449,18 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 			resCfgsPerAPI[copied.API] = append(resCfgsPerAPI[copied.API], &confs[idx])
 			resName := copied.ResourceName()
 
+			// Look up if a resource model registered an attribute map converter. Attribute conversion converts
+			// an untyped, JSON-like object to a typed Go struct. There is a default converter if no
+			// AttributeMapConverter is registered during resource model registration. Lookup will fail for
+			// non-builtin models (so lookup will fail for modular resources) but conversion will happen on the module-side.
 			reg, ok := resource.LookupRegistration(resName.API, copied.Model)
 			if !ok || reg.AttributeMapConverter == nil {
 				continue
 			}
 
 			converted, err := reg.AttributeMapConverter(conf.Attributes)
 			if err != nil {
+				// if conversion errors, the robot will not learn of the new config until it is corrected.
```

Comment:
> the robot will not learn of

What does this mean?

Reply:
updated comment to be a bit clearer, but basically just means that any errors will cause none of the new config to be returned